### ISSUE TITLE
3.3.0 release

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1045]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] run code analysis in VS in *Release* mode and address violations (send a regular PR which must be merged before continuing)
- [x] if necessary, change `VERSION_SUFFIX` in [AppVeyor](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/settings) to  "" and initiate a build
- [x] check [build](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/build/158)
- edit draft release in [GitHub UI](https://github.com/FakeItEasy/FakeItEasy/releases):
    - [x] ensure completeness of release notes, including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
    - [x] attach main nupkg and/or analyzer nupkg, whichever have content to release
    - [x] publish the release
- [x] push nupkgs to NuGet using the Deploy option on the AppVeyor build
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] tweet, mentioning contributors and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [x] [switch to gitflow](https://github.com/FakeItEasy/FakeItEasy/issues/955):
  - [x] create a `develop` branch at the head of `master`
  - [x] apply branch protection to `develop` matching that of `master`
  - [x] make `develop` the default branch
  - [x] send a pull request _to the `develop` branch_ to make appropriate changes to CONTRIBUTING.md
- [ ] create a pull request _to the `develop` branch_ that changes the version in CommonAssemblyInfo.cs to the expected version (of form _xx.yy.zz_)
- [x] create a new draft GitHub Release
- [x] create a new milestone for the next release
- [x] create a new issue (like this one) for the next release, adding it to the new milestone
- [ ] close the milestone for this release
- [x] close this issue

**Comment by thomaslevesque** [comment]
> create a pull request to the develop branch that changes the version in CommonAssemblyInfo.cs to the expected version (of form xx.yy.zz)

Do we still need this if we switch to GitFlow?

**Comment by adamralph** [comment]
Yes, we do. GitFlow is just a branching model. It won't do any versioning magic for us. That will be provided by [GitVersion](https://github.com/FakeItEasy/FakeItEasy/issues/1046).

**Comment by blairconrad** [comment]
But we'll still need to have a step to tag the build to enable that, no?

**Comment by adamralph** [comment]
@blairconrad I'm not sure what you mean.

**Comment by blairconrad** [comment]
GitVersion won't just decide that the version is 3.4.2 or whatever. It'll
still need to be told somehow, like with a tag. No?

**Comment by adamralph** [comment]
Nope, it works off the branches. To being the release process under GitFlow, we'll create a `release-3.4.0` branch from `develop`. A `3.4.0-rc0001` package will be built from the `release-3.4.0` branch (which we may or may not choose to release ahead of RTM). When we want to release RTM, we merge the `release-3.4.0` branch to `master` and that will build the `3.4.0` package. Subsequent builds from `develop` will be `3.5.0-alpha000x`. GitVersion just works it all out auto-magically.

**Comment by blairconrad** [comment]
Ah, okay. But in that case branching and naming the branch release-3.4.0
would fulfill the same purpose as creating the tag, at least as far as
determining the next version goes.
Sounds very slick.

**Comment by blairconrad** [comment]
Tonnes of stuff in 3.3.0 already. I think we're about due for another release.
@adamralph, I think you've the best handle on how we'll want to adopt GitFlow and so on, and want to make the switch right after a version tag is added.
Would a release this week suit?

**Comment by thomaslevesque** [comment]
> Tonnes of stuff in 3.3.0 already. I think we're about due for another release.
> @adamralph, I think you've the best handle on how we'll want to adopt GitFlow and so on, and want to make the switch right after a version tag is added.
> Would a release this week suit?

👍 

We could also include #1051

**Comment by blairconrad** [comment]
Actually, as I read #955, I think even @thomaslevesque or I (sorry to lump you in there with me, @thomaslevesque) could handle the tasks that we'd need to do post release. Let's see how #1051 continues and aim for a release Wednesday or Thursday morning.

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/860076018010390528
https://twitter.com/fakeiteasyfx/status/860075928793362432
